### ISSUE TITLE
GROMACS export crashes due to non-bonded lookup failures when virtual sites are present in some topologies

### ISSUE THREAD

**Issue by megosato** [openforcefield/openff-interchange#1022]
**Description**

Trying to use Interchange and packmol to write out a solvated gromacs topology. 
Using the same code as the minimally reproducible example below but changing out the FF for sage 2.0, 2.1 or 2.2 I have no issues.
I also have no issues if you replace `solv_top = solvate_topology(mol.to_topology())` with just `solv_top = mol.to_topology()`.
I believe the issue has to do with the combination of the virutal sites forcefield and packmol.

**Reproduction**
[example-sigma-hole-bromine.offxml.zip](https://github.com/user-attachments/files/16432814/example-sigma-hole-bromine.offxml.zip)

```
from openff.toolkit import Molecule, ForceField
from openff.interchange.components._packmol import solvate_topology

# create molecule with virtual site
mol = Molecule.from_smiles("CBr")
mol.generate_conformers()

# assign parameters using openff toolkit -> openff interchange

ff = ForceField("openff-2.0.0.offxml", "example-sigma-hole-bromine.offxml")
solv_top = solvate_topology(mol.to_topology())
ic = ff.create_interchange(topology=solv_top)

ic.to_top("cbr.top")
ic.to_gro("cbr.gro")
```

**Output**

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[34], line 3
      1 # write out to gromacs with default 5 nm box
----> 3 ic.to_top("cbr.top")
      4 ic.to_gro("cbr.gro")

File ~/mambaforge/envs/openff-toolkit/lib/python3.12/site-packages/openff/interchange/components/interchange.py:493, in Interchange.to_top(self, file_path, hydrogen_mass, _merge_atom_types)
    489 from openff.interchange.interop.gromacs.export._export import GROMACSWriter
    490 from openff.interchange.smirnoff._gromacs import _convert
    492 GROMACSWriter(
--> 493     system=_convert(self, hydrogen_mass=hydrogen_mass),
    494     top_file=file_path,
    495 ).to_top(_merge_atom_types=_merge_atom_types)

File ~/mambaforge/envs/openff-toolkit/lib/python3.12/site-packages/openff/interchange/smirnoff/_gromacs.py:143, in _convert(interchange, hydrogen_mass)
    140 topology_index = particle_map[interchange.topology.atom_index(atom)]
    141 key = TopologyKey(atom_indices=(topology_index,))
    142 vdw_parameters = vdw_collection.potentials[
--> 143     vdw_collection.key_map[key]
    144 ].parameters
    145 charge = electrostatics_collection.charges[key]
    147 # Build atom types

KeyError: TopologyKey with atom indices (1300,)
```

**Software versions**

- Which operating system and version are you using? 
     - Sonoma 14.2.1 on M1 Mac
- How did you install Interchange?
     - `mamba create -n openff-toolkit -c conda-forge openff-toolkit`
- What is the output of running `conda list`?
```
openeye-toolkits          2024.1.0                py312_0    openeye
openff-amber-ff-ports     0.0.4              pyhca7485f_0    conda-forge
openff-forcefields        2024.07.0          pyhff2d567_0    conda-forge
openff-interchange        0.3.28             pyhd8ed1ab_0    conda-forge
openff-interchange-base   0.3.28             pyhd8ed1ab_0    conda-forge
openff-models             0.1.2              pyhca7485f_0    conda-forge
openff-toolkit            0.16.2             pyhd8ed1ab_0    conda-forge
openff-toolkit-base       0.16.2             pyhd8ed1ab_0    conda-forge
openff-units              0.2.2              pyhca7485f_0    conda-forge
openff-utilities          0.1.12             pyhd8ed1ab_0    conda-forge
openmm                    8.1.2           py312hacce00b_2_khronos    conda-forge
```

**Comment by mattwthompson** [comment]
Thanks for the thorough reproduction - this is legit and I can reproduce this on my machine as well.

**Comment by mattwthompson** [comment]
This should be included in 0.3.29 - please let me know if something looks fishy
